### PR TITLE
Update raft for compatibility with the latest cuco

### PR DESCRIPTION
**Pull request by PointKernel**
This PR updates raft to make it compatible with the latest cuco.

Depends on https://github.com/rapidsai/rapids-cmake/pull/526

CMake changes will be reverted once https://github.com/rapidsai/rapids-cmake/pull/526 is merged.

### CODE REVIEW

File: cpp/include/raft/sparse/distance/detail/coo_spmv_strategies/hash_strategy.cuh
Comment (PointKernel) [reply cpp/include/raft/sparse/distance/detail/coo_spmv_strategies/hash_strategy.cuh:50]: >  is there another similar API we can migrate this to

Yes, we added a more performant [new map](https://github.com/NVIDIA/cuCollections/blob/a992e356c30698224ee89c2e29029425db72825b/include/cuco/static_map.cuh#L88-L97) and it's ready to use. I can put up a PR afterward for the map migration.

**Review by bdice — APPROVED** [review]

**Comment by PointKernel** [comment]
This is ready to go once CI passes. Note the CI will be blocked until this PR gets merged. 

Thanks!

**Comment by PointKernel** [comment]
Can someone please rerun the failed CI task? I didn't see a similar issue in previous CI checks thus maybe it's a random CI runtime error?

**Pull request merged**
